templates/ZtrBoardGame.Console.Tests/StepDefinitions/BoardConnectorStepDefinitions.cs

```
@@ -0,0 +1,84 @@
+    private readonly Mock<HttpMessageHandler> _httpMessageHandlerMock = new();
+    private readonly Mock<ILogger<BoardConnectorService>> _loggerMock = new();
+
+    public BoardConnectorStepDefinitions(ScenarioContext scenarioContext)
+    {
+        _scenarioContext = scenarioContext;
+    }
+
+    [Given(@"PC is running")]
+    public void GivenPcIsRunning()
+    {
+        var services = new ServiceCollection();
+        services.AddSingleton(_loggerMock.Object);
+        services.AddSingleton(_boardStorageMock.Object);
+
+        var httpClient = new HttpClient(_httpMessageHandlerMock.Object);
+        var httpClientFactoryMock = new Mock<IHttpClientFactory>();
+        httpClientFactoryMock.Setup(x => x.CreateClient(It.IsAny<string>())).Returns(httpClient);
+
+        services.AddSingleton(httpClientFactoryMock.Object);
+        services.AddSingleton<BoardConnectorService>();
+        _scenarioContext.Set(services.BuildServiceProvider());
+    }
+
+    [Given(@"board is connected to the PC")]
+    public void GivenBoardIsConnectedToThePc()
+    {
+        _boardStorageMock.Setup(x => x.GetAddresses()).Returns(new List<Uri> { new("http://localhost:5000") });
+    }
+
+    [When(@"board is not available")]
+    public void WhenBoardIsNotAvailable()
+    {
+        _httpMessageHandlerMock.Protected()
+            .Setup<Task<HttpResponseMessage>>(
+                "SendAsync",
+                ItExpr.IsAny<HttpRequestMessage>(),
+                ItExpr.IsAny<CancellationToken>()
+            )
+            .ThrowsAsync(new HttpRequestException("Connection refused"));
+    }
+
+    [Then(@"PC will try to connect to the board")]
+    public async Task ThenPcWillTryToConnectToTheBoard()
+    {
+        var serviceProvider = _scenarioContext.Get<ServiceProvider>();
+        var service = serviceProvider.GetService<BoardConnectorService>();
+        await service.ProcessBoardsAsync(CancellationToken.None);
```

extract method for processing single board. And throw from it and then check the exception instead of logs.